File: backend/src/main/java/com/herd/squire/models/creatures/characters/CharacterPage.java
package com.herd.squire.models.creatures.characters;

public class CharacterPage {
    private CharacterPageType characterPageType;
    private int order;
    private boolean visible;

    public CharacterPage() {}

    public CharacterPage(CharacterPageType characterPageType, int order, boolean visible) {
        this.characterPageType = characterPageType;
        this.order = order;
        this.visible = visible;
    }

    public CharacterPageType getCharacterPageType() {
        return characterPageType;
    }

    public void setCharacterPageType(CharacterPageType characterPageType) {
        this.characterPageType = characterPageType;
    }

    public int getOrder() {
        return order;
    }

    public void setOrder(int order) {
        this.order = order;
    }

    public boolean isVisible() {
        return visible;
    }

    public void setVisible(boolean visible) {
        this.visible = visible;
    }
}

File: frontend/src/app/shared/models/creatures/characters/character-page.ts
import {CharacterPageType} from './character-page-type.enum';

export class CharacterPage {
  characterPageType: CharacterPageType;
  order = 0;
  visible = true;

  constructor(characterPageType: CharacterPageType, order: number, visible: boolean = true) {
    this.characterPageType = characterPageType;
    this.order = order;
    this.visible = visible;
  }
}
